contracts/GenericLender/GenericScream.sol

```diff
@@ -0,0 +1,308 @@
+        return _nav();
+    }
+
+    //adjust dust threshol
+    function setDustThreshold(uint256 amount) external management {
+        dustThreshold = amount;
+    }
+
+    function _nav() internal view returns (uint256) {
+        uint256 amount = want.balanceOf(address(this)).add(underlyingBalanceStored());
+
+        if(amount < dustThreshold){
+            return 0;
+        }else{
+            return amount;
+        }
+    }
+
+    function underlyingBalanceStored() public view returns (uint256 balance) {
+        uint256 currentCr = cToken.balanceOf(address(this));
+        if (currentCr < dustThreshold) {
+            balance = 0;
+        } else {
+            //The current exchange rate as an unsigned integer, scaled by 1e18.
+            balance = currentCr.mul(cToken.exchangeRateStored()).div(1e18);
+        }
+    }
+
+    function apr() external view override returns (uint256) {
+        return _apr();
+    }
+
+    function _apr() internal view returns (uint256) {
+        return (cToken.supplyRatePerBlock().add(compBlockShareInWant(0, false))).mul(blocksPerYear);
+    }
+
+    function compBlockShareInWant(uint256 change, bool add) public view returns (uint256){
+        //comp speed is amount to borrow or deposit (so half the total distribution for want)
+        uint256 distributionPerBlock = ComptrollerI(unitroller).compSpeeds(address(cToken));
+
+        //convert to per dolla
+        uint256 totalSupply = cToken.totalSupply().mul(cToken.exchangeRateStored()).div(1e18);
+        if(add){
+            totalSupply = totalSupply.add(change);
+        }else{
+            totalSupply = totalSupply.sub(change);
+        }
+
+        uint256 blockShareSupply = 0;
+        if(totalSupply > 0){
+            blockShareSupply = distributionPerBlock.mul(1e18).div(totalSupply);
+        }
+
+        uint256 estimatedWant =  priceCheck(scream, address(want),blockShareSupply);
+        uint256 compRate;
+        if(estimatedWant != 0){
+            compRate = estimatedWant.mul(9).div(10); //10% pessimist
+
+        }
+
+        return(compRate);
+    }
+
+    //WARNING. manipulatable and simple routing. Only use for safe functions
+    function priceCheck(address start, address end, uint256 _amount) public view returns (uint256) {
+        if (_amount == 0) {
+            return 0;
+        }
+        address[] memory path = getTokenOutPath(start, end);
+        uint256[] memory amounts = IUniswapV2Router02(spookyRouter).getAmountsOut(_amount, path);
+
+        return amounts[amounts.length - 1];
+    }
+
+    function weightedApr() external view override returns (uint256) {
+        uint256 a = _apr();
+        return a.mul(_nav());
+    }
+
+    function withdraw(uint256 amount) external override management returns (uint256) {
+        return _withdraw(amount);
+    }
+
+    //emergency withdraw. sends balance plus amount to governance
+    function emergencyWithdraw(uint256 amount) external override management {
```

Comment:
should this be governance?

Reply:
Idea for this to be management, so that brain can handle this in an emergency. previously this was strategist, but per your recommendation on the original issue this was updated to management.